core/time/timezone/tzdate.odin

```diff
@@ -0,0 +1,149 @@
+package timezone
+
+import "base:runtime"
+
+import "core:time/datetime"
+import "core:fmt"
+import "core:os"
+import "core:time"
+import "core:path/filepath"
+import "core:strings"
+
+TZ_Context :: struct {
+	db_path:               string,
+	local_name:            string,
+
+	local_region:      ^TZ_Region,
+	regions: map[string]TZ_Region,
+
+	allocator: runtime.Allocator,
+}
+
+tzdb_init :: proc(db_path: string, local_path: string, allocator := context.allocator) -> (tzdb: TZ_Context, success: bool) {
+	path, err := os.absolute_path_from_relative(local_path)
+	if err != nil {
+		return
+	}
+	tzone_base := filepath.base(path)
+
+	tmp_path := filepath.join([]string{path, ".."})
+	cleaned_path := filepath.clean(tmp_path)
+	delete(tmp_path)
+
+	parent_dir := filepath.base(cleaned_path)
+	tzone_name := filepath.join([]string{parent_dir, tzone_base})
+	delete(cleaned_path)
+
+	return TZ_Context{
+		db_path = strings.clone(db_path, allocator), 
+		local_name = tzone_name, 
+		regions = make(map[string]TZ_Region, 8, allocator),
+		allocator = allocator,
+	}, true
+}
+
+tzdb_destroy :: proc(tzdb: ^TZ_Context) {
```

Comment:
This also needs to take an optional allocator to delete things on the proper allocator.